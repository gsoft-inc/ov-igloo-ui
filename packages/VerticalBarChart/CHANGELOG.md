# @igloo-ui/vertical-bar-chart

## 0.3.4

### Patch Changes

- d8dc4ea: Updated Hopper-UI tokens dependencies

## 0.3.3

### Patch Changes

- 81723d0: Updated Hopper depedency

## 0.3.2

### Patch Changes

- 1613653: Updated Hopper Tokens Fonts package

## 0.3.1

### Patch Changes

- 6ea531f: Updated dependency versions for react-aria packages, luxon and hopper

## 0.3.0

### Minor Changes

- 4846c59: Wrapped Igloo CSS in @layer to isolate it.

## 0.2.4

### Patch Changes

- cfb5619: Updated hopper token versions and button versions.

## 0.2.3

### Patch Changes

- 7d36815: Updated hopper token versions and igloo icon versions.

## 0.2.2

### Patch Changes

- a2df55d: Updated to latest Hopper Token dependency

## 0.2.1

### Patch Changes

- 9739155: Updated storybook and recharts. Applied the new brand look for filter.
- 9739155: Gave ColorPicker the new Wokleap look. Color now uses css variables to update its color. Fixed spacing issues in tooltip. Updated hopper token version.

## 0.2.0

### Minor Changes

- 412cbc8: Added the new brand styling to Metric and VerticalBarChart.

## 0.1.0

### Minor Changes

- c3efaf5: The Initial release of the vertical bar chart.
