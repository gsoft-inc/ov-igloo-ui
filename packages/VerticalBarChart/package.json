{
  "name": "@igloo-ui/vertical-bar-chart",
  "version": "0.3.0",
  "main": "dist/VerticalBarChart.js",
  "module": "dist/VerticalBarChart.js",
  "types": "dist/VerticalBarChart.d.ts",
  "styles": "dist/vertical-bar-chart.css",
  "publishConfig": {
    "access": "public"
  },
  "repository": "git+https://github.com/gsoft-inc/ov-igloo-ui.git",
  "author": "Officevibe Inc.",
  "contributors": [
    "Vicky Comeau vicky.comeau@gsoft.com"
  ],
  "license": "Apache-2.0",
  "peerDependencies": {
    "react": ">= 16.8.6"
  },
  "files": [
    "dist"
  ],
  "scripts": {
    "build": "rollup -c rollup.config.js"
  },
  "dependencies": {
    "@igloo-ui/tokens": "^2.0.0",
    "@hopper-ui/tokens": "^3.1.4",
    "classnames": "^2.3.2",
    "recharts": "^2.9.2"
  },
  "browserslist": [
    "> 1%",
    "last 2 versions",
    "IE >= 11"
  ]
}
