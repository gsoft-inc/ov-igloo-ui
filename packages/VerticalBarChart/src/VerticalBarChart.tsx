import * as React from 'react';
import cx from 'classnames';
import {
  BarChart,
  Cell,
  Bar,
  XAxis,
  YAxis,
  ResponsiveContainer,
  CellProps,
  XAxisProps,
  YAxisProps,
} from 'recharts';

import './vertical-bar-chart.scss';

export interface DataProps {
  /** Add a custom class name */
  className?: string;
  /** Add a custom color */
  color?: string;
  /** The score of the bar used for the x-axis */
  score: number;
  /** The value of the bar used for the y-axis */
  value: number;
}

export interface VerticalBarChartProps extends React.ComponentProps<'div'> {
  /** Add a custom class name */
  className?: string;
  /** The data to be displayed in the chart */
  data: DataProps[];
  /** Add a data-test tag for automated tests */
  dataTest?: string;
  /** The height of the chart */
  height?: number;
  /** Whether or not the chart bars should be animated */
  isAnimationActive?: boolean;
  /** The label to be displayed on the bottom right of the chart */
  maxLabel?: string;
  /** The label to be displayed on the bottom left of the chart */
  minLabel?: string;
}

const VerticalBarChart: React.FunctionComponent<VerticalBarChartProps> = ({
  className,
  data,
  dataTest,
  height = 200,
  isAnimationActive = true,
  maxLabel = '10',
  minLabel = '0',
  ...rest
}: VerticalBarChartProps) => {
  const classes = cx('ids-vertical-bar-chart', className);

  const tickProps: React.SVGProps<SVGTextElement> = {
    strokeWidth: 0,
  };
  const xAxisProps: XAxisProps = {
    dataKey: 'score',
    tickLine: false,
    strokeWidth: '1px',
    tick: tickProps,
  };
  const yAxisProps: YAxisProps = {
    type: 'number',
    domain: [0, 'dataMax'],
    hide: true,
  };

  return (
    <div className={classes} data-test={dataTest} {...rest}>
      <ResponsiveContainer width="100%" height={height}>
        <BarChart
          data={data}
          barCategoryGap="15%"
          margin={{ top: 2, left: 2, right: 2, bottom: 2 }}
          width={800}
          /* Width and height here are ignored because of ResponsiveContainer, 
          but are useful for tests. */
          height={height}
        >
          <XAxis {...xAxisProps} />
          <YAxis {...yAxisProps} />
          <Bar
            dataKey="value"
            radius={[5, 5, 0, 0]}
            isAnimationActive={isAnimationActive}
          >
            {data.map((entry, index) => {
              const cellProps: CellProps = {
                className: cx('ids-vertical-bar-chart__bar', entry.className),
                key: `cell-${index}`,
              };
              if (entry.color) {
                cellProps.fill = entry.color;
              }
              return <Cell {...cellProps} />;
            })}
          </Bar>
        </BarChart>
      </ResponsiveContainer>
      <div className="ids-vertical-bar-chart__labels">
        <span className="ids-vertical-bar-chart__min-label">{minLabel}</span>
        <span className="ids-vertical-bar-chart__max-label">{maxLabel}</span>
      </div>
    </div>
  );
};

export default VerticalBarChart;
