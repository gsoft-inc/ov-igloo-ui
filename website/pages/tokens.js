import { useEffect, useState } from 'react';
import { useRouter } from 'next/router';
import cx from 'classnames';

import Link from 'next/link';
import pkg from '@igloo-ui/tokens/package.json';
import Display from '../components/Display';
import Title from '../components/Title';
import Code from '../components/Code';
import Tag from '../components/Tag';
import TokenList from '../components/TokenList';

import iconToken from '../svg/token.svg';

export default function Tokens(props) {
  const { data, categories } = props;
  const { asPath } = useRouter();

  const [activeLink, setActiveLink] = useState();
  const [baseToken, setBaseToken] = useState('10');
  const [styleToken, setStyleToken] = useState('scss');

  const handleTokenChange = (event) => {
    setBaseToken(event.currentTarget.value);
  };

  const handleStyleChange = (event) => {
    setStyleToken(event.currentTarget.value);
  };

  useEffect(() => {
    setActiveLink(window.location.hash);
  }, [asPath]);

  const tokensList = categories.map((key) => {
    const dataType = baseToken === '10' ? 'base10' : 'default';

    const token = data[dataType].filter(
      (obj) => Object.keys(obj).toString() === key
    );
    return (
      <TokenList
        key={key}
        data={token[0][key]}
        title={key}
        options={{ base: baseToken, style: styleToken }}
      />
    );
  });

  const tokensNav = categories.map((category, index) => {
    return (
      <li key={index.toString()}>
        <Link href={`/tokens#${category}`}>
          <a className={cx(activeLink === `#${category}` && 'is-active')}>
            {category}
          </a>
        </Link>
      </li>
    );
  });

  return (
    <>
      <div className="io-section io-section--grid io-section--hasAside">
        <div className="io-aside">
          <ul className="io-subnav">{tokensNav}</ul>
        </div>
        <div className="io-tokens">
          <Display icon={iconToken}>
            <div className="io-display__content">
              Tokens{' '}
              <Tag appearance="ghost" className="io-display__tag">
                v{pkg.version}
              </Tag>
            </div>
          </Display>
          <Title level={2} as="h2">
            Getting started
          </Title>
          <p>
            To install <Code inline>@igloo-ui/tokens</Code> in your project, you
            will need to run the following command using{' '}
            <a href="npmjs.org" target="_blank">
              npm:
            </a>
          </p>
          <Code language="bash">{`npm install @igloo-ui/icons`}</Code>
          <p>If you prefer Yarn, </p>
          <Code language="bash">{`yarn add @igloo-ui/icons`}</Code>
          <Title level={3} as="h3">
            Usage
          </Title>
          <p>In Sass, you can import the files by doing:</p>
          <Code language="jsx">
            {`// Fonts${'\n'}@import '@igloo-ui/tokens/dist/_fonts.scss';${'\n'}// SCSS${'\n'}@import '@igloo-ui/tokens/dist/_variables.scss';${'\n'}// CSS${'\n'}@import '@igloo-ui/tokens/dist/_colors.scss';${'\n'}`}
          </Code>
          <Title level={3} as="h3">
            Base 10 support
          </Title>
          <p>
            If your project uses the mathematical trick of basing the value of{' '}
            <Code inline>1rem</Code> equals <Code inline>10px</Code>, there is
            also an <strong>base10</strong> output within the dist folder of
            this package.
          </p>
          <p>In CSS, you can import the files by doing:</p>
          <Code>@import '~@igloo-ui/tokens/dist/base10/variables.css';</Code>

          <div className="io-tokens__list">
            <div className="io-options io-options--large">
              <Title level={3} as="h3" className="io-options__title">
                Options
              </Title>
              <select
                className="io-select"
                value={styleToken}
                onChange={handleStyleChange}
              >
                <option selected value="scss">
                  scss
                </option>
                <option value="css">css</option>
              </select>
              <select
                className="io-select"
                value={baseToken}
                onChange={handleTokenChange}
              >
                <option selected value="10">
                  base 10
                </option>
                <option value="16">base 16</option>
              </select>
            </div>
            {tokensList}
          </div>
        </div>
      </div>
    </>
  );
}

export async function getStaticProps() {
  const fetcher = (path) =>
    fetch(
      `https://raw.githubusercontent.com/gsoft-inc/ov-igloo-tokens/main${path}`
    ).then((res) => res.json());

  const tokens = await fetcher('/docs/tokens.json');
  const tokensBase10 = await fetcher('/docs/base10/tokens.json');

  const categories = tokens.map((obj) => Object.keys(obj)).flat(1);

  if (!tokens || !tokensBase10) {
    return {
      notFound: true,
    };
  }

  return {
    props: { data: { default: tokens, base10: tokensBase10 }, categories }, // will be passed to the page component as props
  };
}
