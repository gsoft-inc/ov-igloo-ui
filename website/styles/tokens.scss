@use "base/colors" as color;
@use "abstract/function" as *;

// active color theme
$active-theme: color.$light-theme;

//space tokens
$space-0: var(--space-0);
$space-1: var(--space-1);
$space-2: var(--space-2);
$space-3: var(--space-3);
$space-4: var(--space-4);
$space-5: var(--space-5);
$space-6: var(--space-6);
$space-7: var(--space-7);
$space-8: var(--space-8);
$space-9: var(--space-9);
$space-10: var(--space-10);

//color tokens
$grey-100: var(--neutral-100);
$grey-200: var(--neutral-200);
$grey-300: var(--neutral-300);
$grey-400: var(--neutral-400);
$grey-500: var(--neutral-500);
$grey-600: var(--neutral-600);
$grey-700: var(--neutral-700);
$grey-800: var(--neutral-800);
$grey-900: var(--neutral-900);
$white: var(--neutral-0);
$black: var(--neutral-1000);
$green: #09a198;
$yellow: #ffc737;
$pink: #ff8ea9;

//radius tokens
$radius-0: var(--space-0);
$radius-1: rebase(0.5rem);
$radius-2: rebase(0.25rem);

// layout tokens
$layout-medium: rebase(80rem);
$layout-small: rebase(50rem);

// Semantic colors
$text-color: $black;
$bg-color: $white;
$body-bg-color: $grey-100;

$navbar-bg-color: $white;
$navbar-color: $black;
$navbar-min-height: var(--space-9);

$hero-space: $space-10 0 rebase(8rem) 0;

$menu-bg-color: $white;
$menu-color: $black;
$menu-border-color: $black;
$menu-border: 1px solid $menu-border-color;
$menu-text-size: 1.6rem;
$menu-radius: $radius-0;
$menu-shadow: 0.5px 1.6px 1.6px rgb(0 0 0 / 15%);
$code-bg-color: var(--code-300);
$code-color: var(--code-700);
$button-copy-border: rgba(205, 217, 229, 0.2);
$button-copy-color: #a7adb3;

// Semantic font sizes
$text-size: var(--fs-300);
$text-size-small: var(--fs-100);

$max-layout-width: $layout-medium;
