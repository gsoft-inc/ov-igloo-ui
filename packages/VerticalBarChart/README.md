# VerticalBarChart

A vertical bar chart is a graphical representation that uses vertical bars to display data values, with each bar representing a category or group. The height of each bar corresponds to the magnitude of the data it represents, allowing for easy comparison between different categories.

<Example is="custom" />

<ReferenceLinks is="custom" />

## Installation

To install `@igloo-ui/vertical-bar-chart` in your project, you will need to run the following command using [npm](https://www.npmjs.com/):

```bash
npm install @igloo-ui/vertical-bar-chart
```

If you prefer [Yarn](https://classic.yarnpkg.com/en/), use the following command instead:

```bash
yarn add @igloo-ui/vertical-bar-chart
```

## Usage

Then to use the component in your code just import it!

```jsx
import VerticalBarChart from '@igloo-ui/vertical-bar-chart';
import tokens from '@igloo-ui/tokens/dist/base10/tokens.json';

const getColorFromScore = (score) => {
  if (score >= 0 && score < 2) return tokens.coral400;
  if (score >= 2 && score < 4) return tokens.coral200;
  if (score >= 4 && score < 6) return tokens.creamsicle100;
  if (score >= 6 && score < 9) return tokens.seaweed200;
  return tokens.seaweed400;
};

const values = [12, 20, 35, 18, 3, 42, 60, 0, 16, 43, 15];
const chartData = values.map((value, index) => {
  return {
    score: index,
    value,
    color: getColorFromScore(index),
  };
});

<VerticalBarChart data={chartData} />;
```
