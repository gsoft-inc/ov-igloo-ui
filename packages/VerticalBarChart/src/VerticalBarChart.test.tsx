/**
 * @jest-environment jsdom
 */
import React from 'react';
import { render, screen } from '@testing-library/react';

import tokens from '@igloo-ui/tokens/dist/base10/tokens.json';
import VerticalBarChart, { VerticalBarChartProps } from './VerticalBarChart';

jest.mock('recharts', () => {
  const OriginalRechartsModule = jest.requireActual('recharts');

  return {
    ...OriginalRechartsModule,
    ResponsiveContainer: ({ height, children }: any) => (
      <div className="recharts-responsive-container" style={{ width: 800, height }}>
        {children}
      </div>
    ),
  };
});

const getColorFromScore = (score: number) => {
  if (score >= 0 && score < 2) return tokens.coral400;
  if (score >= 2 && score < 4) return tokens.coral200;
  if (score >= 4 && score < 6) return tokens.creamsicle100;
  if (score >= 6 && score < 9) return tokens.seaweed200;
  return tokens.seaweed400;
};

const values = [12, 20, 35, 18, 3, 42, 60, 33, 16, 43, 15];
const chartData = values.map((value, index) => {
  return {
    score: index,
    value,
    color: getColorFromScore(index),
  };
});

const chartDefaults = {
  data: chartData,
  isAnimationActive: false,
};

const setup = (
  props: VerticalBarChartProps = chartDefaults
) => {
  return render(
    <VerticalBarChart dataTest="ids-vertical-bar-chart" {...props} />
  );
};

describe('VerticalBarChart', () => {
  test('It should render without errors', () => {
    setup();
    const wrapper = screen.getByTestId('ids-vertical-bar-chart');
    expect(wrapper).toBeInTheDocument();
  });

  test('It should render a snapshot', () => {
    const { asFragment } = setup();
    expect(asFragment()).toMatchSnapshot();
  });

  test('It should render the provided data elements', () => {
    const { container } = setup();
    const dataElements = container.querySelectorAll(
      '.ids-vertical-bar-chart__bar'
    );
    expect(dataElements.length).toBe(chartData.length);
  });

  test('It should render the correct height', () => {
    setup({height: 600, ...chartDefaults});
    const chartContainer = screen.getByRole('region');
    const containerStyles = window.getComputedStyle(chartContainer);
    const containerHeight = containerStyles.height;
    expect(containerHeight).toBe('600px');
  });

  test('It should render the max and min labels', () => {
    const minLabelText = 'Negative Label';
    const maxLabelText = 'Positive Label';
    setup({minLabel: minLabelText, maxLabel: maxLabelText, ...chartDefaults});
    const minLabel = screen.getByText(minLabelText);
    const maxLabel = screen.getByText(maxLabelText);
    expect(minLabel).toBeInTheDocument();
    expect(maxLabel).toBeInTheDocument();
  });
});
