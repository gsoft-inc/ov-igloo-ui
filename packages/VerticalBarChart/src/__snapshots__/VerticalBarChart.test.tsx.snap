// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VerticalBarChart It should render a snapshot 1`] = `
<DocumentFragment>
  <div
    class="ids-vertical-bar-chart"
    data-test="ids-vertical-bar-chart"
  >
    <div
      class="recharts-responsive-container"
      style="width: 800px; height: 200px;"
    >
      <div
        class="recharts-wrapper"
        role="region"
        style="position: relative; cursor: default; width: 800px; height: 200px;"
      >
        <svg
          class="recharts-surface"
          height="200"
          viewBox="0 0 800 200"
          width="800"
        >
          <title />
          <desc />
          <defs>
            <clippath
              id="recharts4-clip"
            >
              <rect
                height="166"
                width="796"
                x="2"
                y="2"
              />
            </clippath>
          </defs>
          <g
            class="recharts-layer recharts-cartesian-axis recharts-xAxis xAxis"
          >
            <line
              class="recharts-cartesian-axis-line"
              fill="none"
              height="30"
              orientation="bottom"
              stroke="#666"
              stroke-width="1px"
              width="796"
              x="2"
              x1="2"
              x2="798"
              y="168"
              y1="168"
              y2="168"
            />
            <g
              class="recharts-cartesian-axis-ticks"
            >
              <g
                class="recharts-layer recharts-cartesian-axis-tick"
              >
                <text
                  class="recharts-text recharts-cartesian-axis-tick-value"
                  fill="#666"
                  height="30"
                  orientation="bottom"
                  stroke="none"
                  stroke-width="0"
                  text-anchor="middle"
                  width="796"
                  x="38.18181818181818"
                  y="176"
                >
                  <tspan
                    dy="0.71em"
                    x="38.18181818181818"
                  >
                    0
                  </tspan>
                </text>
              </g>
              <g
                class="recharts-layer recharts-cartesian-axis-tick"
              >
                <text
                  class="recharts-text recharts-cartesian-axis-tick-value"
                  fill="#666"
                  height="30"
                  orientation="bottom"
                  stroke="none"
                  stroke-width="0"
                  text-anchor="middle"
                  width="796"
                  x="110.54545454545453"
                  y="176"
                >
                  <tspan
                    dy="0.71em"
                    x="110.54545454545453"
                  >
                    1
                  </tspan>
                </text>
              </g>
              <g
                class="recharts-layer recharts-cartesian-axis-tick"
              >
                <text
                  class="recharts-text recharts-cartesian-axis-tick-value"
                  fill="#666"
                  height="30"
                  orientation="bottom"
                  stroke="none"
                  stroke-width="0"
                  text-anchor="middle"
                  width="796"
                  x="182.9090909090909"
                  y="176"
                >
                  <tspan
                    dy="0.71em"
                    x="182.9090909090909"
                  >
                    2
                  </tspan>
                </text>
              </g>
              <g
                class="recharts-layer recharts-cartesian-axis-tick"
              >
                <text
                  class="recharts-text recharts-cartesian-axis-tick-value"
                  fill="#666"
                  height="30"
                  orientation="bottom"
                  stroke="none"
                  stroke-width="0"
                  text-anchor="middle"
                  width="796"
                  x="255.27272727272725"
                  y="176"
                >
                  <tspan
                    dy="0.71em"
                    x="255.27272727272725"
                  >
                    3
                  </tspan>
                </text>
              </g>
              <g
                class="recharts-layer recharts-cartesian-axis-tick"
              >
                <text
                  class="recharts-text recharts-cartesian-axis-tick-value"
                  fill="#666"
                  height="30"
                  orientation="bottom"
                  stroke="none"
                  stroke-width="0"
                  text-anchor="middle"
                  width="796"
                  x="327.6363636363636"
                  y="176"
                >
                  <tspan
                    dy="0.71em"
                    x="327.6363636363636"
                  >
                    4
                  </tspan>
                </text>
              </g>
              <g
                class="recharts-layer recharts-cartesian-axis-tick"
              >
                <text
                  class="recharts-text recharts-cartesian-axis-tick-value"
                  fill="#666"
                  height="30"
                  orientation="bottom"
                  stroke="none"
                  stroke-width="0"
                  text-anchor="middle"
                  width="796"
                  x="400"
                  y="176"
                >
                  <tspan
                    dy="0.71em"
                    x="400"
                  >
                    5
                  </tspan>
                </text>
              </g>
              <g
                class="recharts-layer recharts-cartesian-axis-tick"
              >
                <text
                  class="recharts-text recharts-cartesian-axis-tick-value"
                  fill="#666"
                  height="30"
                  orientation="bottom"
                  stroke="none"
                  stroke-width="0"
                  text-anchor="middle"
                  width="796"
                  x="472.3636363636363"
                  y="176"
                >
                  <tspan
                    dy="0.71em"
                    x="472.3636363636363"
                  >
                    6
                  </tspan>
                </text>
              </g>
              <g
                class="recharts-layer recharts-cartesian-axis-tick"
              >
                <text
                  class="recharts-text recharts-cartesian-axis-tick-value"
                  fill="#666"
                  height="30"
                  orientation="bottom"
                  stroke="none"
                  stroke-width="0"
                  text-anchor="middle"
                  width="796"
                  x="544.7272727272726"
                  y="176"
                >
                  <tspan
                    dy="0.71em"
                    x="544.7272727272726"
                  >
                    7
                  </tspan>
                </text>
              </g>
              <g
                class="recharts-layer recharts-cartesian-axis-tick"
              >
                <text
                  class="recharts-text recharts-cartesian-axis-tick-value"
                  fill="#666"
                  height="30"
                  orientation="bottom"
                  stroke="none"
                  stroke-width="0"
                  text-anchor="middle"
                  width="796"
                  x="617.090909090909"
                  y="176"
                >
                  <tspan
                    dy="0.71em"
                    x="617.090909090909"
                  >
                    8
                  </tspan>
                </text>
              </g>
              <g
                class="recharts-layer recharts-cartesian-axis-tick"
              >
                <text
                  class="recharts-text recharts-cartesian-axis-tick-value"
                  fill="#666"
                  height="30"
                  orientation="bottom"
                  stroke="none"
                  stroke-width="0"
                  text-anchor="middle"
                  width="796"
                  x="689.4545454545454"
                  y="176"
                >
                  <tspan
                    dy="0.71em"
                    x="689.4545454545454"
                  >
                    9
                  </tspan>
                </text>
              </g>
              <g
                class="recharts-layer recharts-cartesian-axis-tick"
              >
                <text
                  class="recharts-text recharts-cartesian-axis-tick-value"
                  fill="#666"
                  height="30"
                  orientation="bottom"
                  stroke="none"
                  stroke-width="0"
                  text-anchor="middle"
                  width="796"
                  x="761.8181818181818"
                  y="176"
                >
                  <tspan
                    dy="0.71em"
                    x="761.8181818181818"
                  >
                    10
                  </tspan>
                </text>
              </g>
            </g>
          </g>
          <g
            class="recharts-layer recharts-bar"
          >
            <g
              class="recharts-layer recharts-bar-rectangles"
            >
              <g
                class="recharts-layer recharts-bar-rectangle"
              >
                <path
                  class="recharts-rectangle ids-vertical-bar-chart__bar"
                  color="#FF7160"
                  d="M12.854545454545455,139.8A 5,5,0,0,1,17.854545454545455,134.8L 57.85454545454546,134.8A 5,5,0,0,1,
        62.85454545454546,139.8L 62.85454545454546,168L 12.854545454545455,168Z"
                  fill="#FF7160"
                  height="33.19999999999999"
                  radius="5,5,0,0"
                  width="50"
                  x="12.854545454545455"
                  y="134.8"
                />
              </g>
              <g
                class="recharts-layer recharts-bar-rectangle"
              >
                <path
                  class="recharts-rectangle ids-vertical-bar-chart__bar"
                  color="#FF7160"
                  d="M85.21818181818182,117.66666666666669A 5,5,0,0,1,90.21818181818182,112.66666666666669L 130.21818181818182,112.66666666666669A 5,5,0,0,1,
        135.21818181818182,117.66666666666669L 135.21818181818182,168L 85.21818181818182,168Z"
                  fill="#FF7160"
                  height="55.333333333333314"
                  radius="5,5,0,0"
                  width="50"
                  x="85.21818181818182"
                  y="112.66666666666669"
                />
              </g>
              <g
                class="recharts-layer recharts-bar-rectangle"
              >
                <path
                  class="recharts-rectangle ids-vertical-bar-chart__bar"
                  color="#FFC3BA"
                  d="M157.58181818181816,76.16666666666667A 5,5,0,0,1,162.58181818181816,71.16666666666667L 202.58181818181816,71.16666666666667A 5,5,0,0,1,
        207.58181818181816,76.16666666666667L 207.58181818181816,168L 157.58181818181816,168Z"
                  fill="#FFC3BA"
                  height="96.83333333333333"
                  radius="5,5,0,0"
                  width="50"
                  x="157.58181818181816"
                  y="71.16666666666667"
                />
              </g>
              <g
                class="recharts-layer recharts-bar-rectangle"
              >
                <path
                  class="recharts-rectangle ids-vertical-bar-chart__bar"
                  color="#FFC3BA"
                  d="M229.9454545454545,123.19999999999999A 5,5,0,0,1,234.9454545454545,118.19999999999999L 274.9454545454545,118.19999999999999A 5,5,0,0,1,
        279.9454545454545,123.19999999999999L 279.9454545454545,168L 229.9454545454545,168Z"
                  fill="#FFC3BA"
                  height="49.80000000000001"
                  radius="5,5,0,0"
                  width="50"
                  x="229.9454545454545"
                  y="118.19999999999999"
                />
              </g>
              <g
                class="recharts-layer recharts-bar-rectangle"
              >
                <path
                  class="recharts-rectangle ids-vertical-bar-chart__bar"
                  color="#FFE3C2"
                  d="M302.3090909090909,163.85A 4.150000000000006,4.150000000000006,0,0,1,306.45909090909095,159.7L 348.1590909090909,159.7A 4.150000000000006,4.150000000000006,0,0,1,
        352.3090909090909,163.85L 352.3090909090909,168L 302.3090909090909,168Z"
                  fill="#FFE3C2"
                  height="8.300000000000011"
                  radius="5,5,0,0"
                  width="50"
                  x="302.3090909090909"
                  y="159.7"
                />
              </g>
              <g
                class="recharts-layer recharts-bar-rectangle"
              >
                <path
                  class="recharts-rectangle ids-vertical-bar-chart__bar"
                  color="#FFE3C2"
                  d="M374.6727272727273,56.800000000000004A 5,5,0,0,1,379.6727272727273,51.800000000000004L 419.6727272727273,51.800000000000004A 5,5,0,0,1,
        424.6727272727273,56.800000000000004L 424.6727272727273,168L 374.6727272727273,168Z"
                  fill="#FFE3C2"
                  height="116.19999999999999"
                  radius="5,5,0,0"
                  width="50"
                  x="374.6727272727273"
                  y="51.800000000000004"
                />
              </g>
              <g
                class="recharts-layer recharts-bar-rectangle"
              >
                <path
                  class="recharts-rectangle ids-vertical-bar-chart__bar"
                  color="#9FE9DC"
                  d="M447.0363636363636,7A 5,5,0,0,1,452.0363636363636,2L 492.0363636363636,2A 5,5,0,0,1,
        497.0363636363636,7L 497.0363636363636,168L 447.0363636363636,168Z"
                  fill="#9FE9DC"
                  height="166"
                  radius="5,5,0,0"
                  width="50"
                  x="447.0363636363636"
                  y="2"
                />
              </g>
              <g
                class="recharts-layer recharts-bar-rectangle"
              >
                <path
                  class="recharts-rectangle ids-vertical-bar-chart__bar"
                  color="#9FE9DC"
                  d="M519.4,81.69999999999999A 5,5,0,0,1,524.4,76.69999999999999L 564.4,76.69999999999999A 5,5,0,0,1,
        569.4,81.69999999999999L 569.4,168L 519.4,168Z"
                  fill="#9FE9DC"
                  height="91.30000000000001"
                  radius="5,5,0,0"
                  width="50"
                  x="519.4"
                  y="76.69999999999999"
                />
              </g>
              <g
                class="recharts-layer recharts-bar-rectangle"
              >
                <path
                  class="recharts-rectangle ids-vertical-bar-chart__bar"
                  color="#9FE9DC"
                  d="M591.7636363636364,128.73333333333335A 5,5,0,0,1,596.7636363636364,123.73333333333335L 636.7636363636364,123.73333333333335A 5,5,0,0,1,
        641.7636363636364,128.73333333333335L 641.7636363636364,168L 591.7636363636364,168Z"
                  fill="#9FE9DC"
                  height="44.26666666666665"
                  radius="5,5,0,0"
                  width="50"
                  x="591.7636363636364"
                  y="123.73333333333335"
                />
              </g>
              <g
                class="recharts-layer recharts-bar-rectangle"
              >
                <path
                  class="recharts-rectangle ids-vertical-bar-chart__bar"
                  color="#30C9AD"
                  d="M664.1272727272727,54.03333333333333A 5,5,0,0,1,669.1272727272727,49.03333333333333L 709.1272727272727,49.03333333333333A 5,5,0,0,1,
        714.1272727272727,54.03333333333333L 714.1272727272727,168L 664.1272727272727,168Z"
                  fill="#30C9AD"
                  height="118.96666666666667"
                  radius="5,5,0,0"
                  width="50"
                  x="664.1272727272727"
                  y="49.03333333333333"
                />
              </g>
              <g
                class="recharts-layer recharts-bar-rectangle"
              >
                <path
                  class="recharts-rectangle ids-vertical-bar-chart__bar"
                  color="#30C9AD"
                  d="M736.4909090909091,131.5A 5,5,0,0,1,741.4909090909091,126.5L 781.4909090909091,126.5A 5,5,0,0,1,
        786.4909090909091,131.5L 786.4909090909091,168L 736.4909090909091,168Z"
                  fill="#30C9AD"
                  height="41.5"
                  radius="5,5,0,0"
                  width="50"
                  x="736.4909090909091"
                  y="126.5"
                />
              </g>
            </g>
            <g
              class="recharts-layer"
            />
          </g>
        </svg>
      </div>
    </div>
    <div
      class="ids-vertical-bar-chart__labels"
    >
      <span
        class="ids-vertical-bar-chart__min-label"
      >
        0
      </span>
      <span
        class="ids-vertical-bar-chart__max-label"
      >
        10
      </span>
    </div>
  </div>
</DocumentFragment>
`;
